posts/how-to-commit-code-to-a-git-repository-with-visual-studio-code.md

```diff
@@ -0,0 +1,172 @@
+    <figcaption>The actions revealed when hovering over the <strong>Changes</strong> title, from left to right: <strong>Discard All Changes</strong> and <strong>Stage All Changes</strong>, followed by the number of changed files which is 3.</figcaption>
+</figure>
+
+
+After clicking the **Stage All Changes** button, all changed files in the list will be added to the staging area. This is a faster option than adding each file one by one.
+
+## Step 4 — Checking the list of staged changes
+
+After staging some changes, a new section will appear in the **Source Control** panel: **Staged Changes**.
+
+For example, after staging `README.md` and `getting-started.txt`, both files will now be present in the **Staged Changes** section and will no longer appear in the **Changes** section.
+
+<figure className="small">
+    <img src="https://storage.googleapis.com/sourcegraph-assets/learn/tutorial-images/vs-code-source-control-staged-changes.png" alt="The Staged Changes showing that the `README.md` file is staged." />
+    <figcaption>The <strong>Staged Changes</strong> showing that the <code>README.md</code> file is staged.</figcaption>
+</figure>
+
+If any changes remain in the **Changes** section when we commit, they will remain in the working directory and will be available to be staged for a future commit.
+
+After checking that **Staged Changes** contains all the changes that we intend to commit, we can proceed to write a commit message.
+
+## Step 5 — Writing a commit message and committing
+
+Before we can commit our changes, we need to write a commit message. The commit message should describe the change being made. Commit messages are helpful when we’re browsing the history of a repository and trying to understand changes that were made in the past.
+
+For example, we can write the following commit message for our changed files:
+
+```
+Update Readme and Getting Started docs
+```
+
+We’ll use the **Message** input box at the top of the **Source Control** panel to enter this commit message.
+
+<figure className="small">
+    <img src="https://storage.googleapis.com/sourcegraph-assets/learn/tutorial-images/vs-code-source-control-staged-changes-with-message.png" alt="Writing a commit message in the Message input box." />
+    <figcaption>Writing a commit message in the <strong>Message</strong> input box.</figcaption>
+</figure>
+
+A common convention in Git is to limit the length of commit messages to a maximum of 50 characters. VS Code helps us follow this convention by showing a warning message if we exceed that length.
+
+We’ll use the **Commit** button located at the very top of the **Source Control** panel to create the commit. This button appears as a checkmark icon, visible in the following screenshot:
```

Comment:
Optimize for accessible language, "This button appears as a check mark icon, as in the screenshot below."